src/script/common/c_content.cpp

```diff
@@ -2395,14 +2417,9 @@ void push_hud_element(lua_State *L, HudElement *elem)
 bool read_hud_change(lua_State *L, HudElementStat &stat, HudElement *elem, void **value)
 {
 	std::string statstr = lua_tostring(L, 3);
-	{
-		int statint;
-		if (!string_to_enum(es_HudElementStat, statint, statstr)) {
-			script_log_unique(L, "Unknown HUD stat type: " + statstr, warningstream);
-			return false;
-		}
-
-		stat = static_cast<HudElementStat>(statint);
+	if (!string_to_enum(es_HudElementStat, stat, statstr)) {
+		script_log_unique(L, "Unknown HUD stat type: " + statstr, warningstream);
```

Comment:
Perhaps include the `elem` name in the warning message too?

Reply:
I think this should be enough for the modder actually, and if not there's a backtrace on infostream.